tunnel-server/src/ssh-server.ts

```diff
@@ -4,23 +4,34 @@ import net from 'net'
 import path from 'path'
 import ssh2, { ParsedKey, SocketBindInfo } from 'ssh2'
 import { inspect } from 'util'
+import * as z from 'zod'
 
 const idFromPublicSsh = (key: Buffer) =>
   crypto.createHash('sha1').update(key).digest('base64url').replace(/[_-]/g, '').slice(0, 8).toLowerCase()
 
-const getRequestedSocketPath = (info: ssh2.SocketBindInfo) => {
-  const [path, access] = info.socketPath.split('#')
-  if (access === 'private'){
-    return {
-      path,
-      access: 'private' as const
-    }
-  }
-  return {
-    path,
-    access: 'public' as const
+const schema = z.object({
+  path: z.string(),
+  params: z.object({ access: z.enum(['public', 'private']).default('public') }).required(),
+})
+
+export const parseRequestedPath = (requestPath: string) => {
+  const [path, params] = requestPath.split('#')
+
+  if (!params) {
+    return schema.safeParse({ path, params: {} })
```

Comment:
not sure this is worth a schema validator, as most of the parsing is done via custom code anyways

Reply:
the parsing applies defaults where's needed.